### PR TITLE
Updates to readme about documentation, communication & license.

### PR DESCRIPTION
**Pull request by tfenne**
### Description

Updated the readme to include more information on licensing, updated links to documentation and suggestions on how to interact with the project.

### Checklist

- [ ] Code compiles correctly
- [ ] New tests covering changes and new functionality
- [ ] All tests passing
- [ ] Extended the README / documentation, if necessary
- [ ] Is not backward compatible (breaks binary or source compatibility)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,11 +16,19 @@ common file formats, such as [SAM][1] and [VCF][2], used for high-throughput
...
 manipulating HTS data.
 
-Please see the [HTSJDK Documentation](http://samtools.github.io/htsjdk) for more information.
-
 > **NOTE: _HTSJDK does not currently support the latest Variant Call Format Specification (VCFv4.3 and BCFv2.2)._**
 
-#### Building HTSJDK
+### Documentation & Getting Help
+
+API documentation for all versions of HTSJDK since `1.128` are available through [javadoc.io](http://www.javadoc.io/doc/com.github.samtools/htsjdk).
```
Comment (lbergelson) [inline README.md]: That's neat, I didn't know about that service.

File: README.md [hunk 2]
```diff
@@ -83,11 +91,17 @@ To create a project in IntelliJ IDE for htsjdk do the following:
 
 From time to time if dependencies change in htsjdk you may need to refresh the project from the `View -> Gradle` menu.
 
-#### Licensing Information
+### Licensing Information
+
+Not all sub-packages of htsjdk are subject to the same license, so a license notice is included in each source file or sub-package as appropriate. 
+Please check the relevant license notice whenever you start working with a part of htsjdk that you have not previously worked with to avoid any surprises. 
+Broadly speaking the majority of the code is covered under the MIT license with the following notable exceptions:
```
Comment (lbergelson) [inline README.md]: This is a gross mess.  Hopefully when Broad legal gets back from vacation we can resolve at least the lgpl bits.

File: README.md [hunk 3]
```diff
@@ -83,11 +91,17 @@ To create a project in IntelliJ IDE for htsjdk do the following:
...
 
-Not all sub-packages of htsjdk are subject to the same license, so a license notice is included in each source file or sub-package as appropriate. Please check the relevant license notice whenever you start working with a part of htsjdk that you have not previously worked with to avoid any surprises. 
+* Much of the CRAM code is under the Apache License, Version 2
+* Core `tribble` code (underlying VCF reading/writing amongst other things) is under LGPL
+* Code supporting the reading/writing of SRA format is uncopyrighted & public domain
 
-#### Java Minimum Version Support Policy
+### Java Minimum Version Support Policy
 
 > **NOTE: _Effective November 24th 2015, HTSJDK has ended support of Java 7 and previous versions. Java 8 is now required_.**
```
Comment (lbergelson) [inline README.md]: Should we remove this notice?  It's been around for a long time now.  Anyone who cares probably knows.

Comment (tfenne) [inline README.md]: I removed the note and updated the proposed data for Java 8 to be one our from this month as that's _probably_ when Oracle will stop support for Java 8 (one year from the next major release).

**Comment by tfenne** [comment]
@jacarey Do you want to review this?  How about I merge tomorrow if I don't hear from you, and you can just say "yes I want to review" to postpone that?

**Comment by pshapiro4broad** [comment]
@tfenne 
> @jacarey Do you want to review this? How about I merge tomorrow if I don't hear from you, and you can just say "yes I want to review" to postpone that?

@jacarey is out today so that probably isn't a great strategy.

**Comment by tfenne** [comment]
@pshapiro4broad I hear you ... although in fairness this PR is two weeks old ;)  I'll wait until Wednesday.

**Comment by jacarey** [comment]
@tfenne I'll review it today. Thanks!

**Pull request merged**